data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/OpenSearchSink.java

```diff
@@ -258,7 +260,21 @@ public void doOutput(final Collection<Record<Event>> records) {
   private SerializedJson getDocument(final Event event) {
     String docId = (documentIdField != null) ? event.get(documentIdField, String.class) : null;
     String routing = (routingField != null) ? event.get(routingField, String.class) : null;
-    return SerializedJson.fromStringAndOptionals(event.toJsonString(), docId, routing);
+
+    final String document = buildRawDocument(event);
+
+    return SerializedJson.fromStringAndOptionals(document, docId, routing);
+  }
+
+  private String buildRawDocument(final Event event) {
```

Comment:
Can we have a unit test for this? Or worst case, a new integration test.

Reply:
I can pull this out into an integ test. The way the currect class is written is difficult to unit test.

Reply:
I elected to create a smaller helper class to unit test this logic more thoroughly. I also built an integ test as well that demonstrates this feature is supported as a whole.